src/parser.rs

```diff
@@ -497,6 +500,42 @@ mod tests {
     use {Parser, Value, Section};
     use std::collections::BTreeMap;
 
+    #[test]
+    fn finish_string() {
+        let mut p = Parser::new("\"foObar\"");
+        assert_eq!(Some("foObar"), p.finish_string().unwrap().as_str());
+
+        let mut p = Parser::new("\"foObar");
+        assert_eq!(Some("foObar"), p.finish_string().unwrap().as_str());
+
+        let mut p = Parser::new("");
+        assert_eq!(None, p.finish_string());
+    }
+
+    #[test]
+    fn finish_array() {
+        let mut p = Parser::new("[\"a\"");
+        assert_eq!(None, p.finish_array());
+
+        let mut p = Parser::new("[\"a\"]");
+        assert_eq!(Some(Value::new_string_array("a")), p.finish_array());
+    }
+
+    #[test]
+    fn finish_dictionary() {
```

Comment:
I would add `Parser::new("{}")`, `Parser::new("{")`, `Parser::new("{ foo")` and `Parser::new("{ foo =")` cases